eng/Versions.props

```diff
@@ -45,9 +45,11 @@
     <ProjectServicingConfiguration Include="Microsoft.NETCore.App.Ref" PatchVersion="0" />
   </ItemGroup>
   <PropertyGroup>
+    <MicrosoftCodeAnalysisAnalyzersVersion>3.3.1</MicrosoftCodeAnalysisAnalyzersVersion> 
```

Comment:
Maybe these belong in their own section with a comment saying why these numbers were chosen, and what to be aware of when modifying them.

Typically, it is beneficial to update to the latest versions, but these are sensitive because they mean our source generators may not run with older compilers.

Reply:
Yeah I think since these are build only assets we should target the lowest possible version (the one that started supporting source generators) for our source generators, unless we depend on a new Roslyn API for the generator.